### PR TITLE
[TD-6] more structure for junit documentation

### PR DESCRIPTION
**Pull request by circlecai**
# Description
* Organize test runners by language
* TOC with the languages
* Table with summary of languages, test runners, formatters, and links to
example/docs
* Added a couple more things (how to use pytest to produce test results for
unittest, kaocha runner for clojure)

# Reasons
https://circleci.atlassian.net/browse/TD-6

### CODE REVIEW

**Comment by ccgeller** [comment]
Maybe this PR has to be updated with master but I made a [preview link ](http://circleci-doc-preview.s3-website-us-east-1.amazonaws.com/junit-structure-preview/2.0/collect-test-data/#enabling-formatters) And it seems the side nav is no longer highlighting on scroll 
You can see on the current site 'Metadata collection in custom test steps' is highlighted [link](https://circleci.com/docs/2.0/collect-test-data/) but in the preview for this PR the side TOC stops highlighting
Screen Shot 2022-01-04 at 2 54 37 PM

**Comment by circlecai** [comment]
Sorry about that. I've rebased and it is now working on my local dev.

File: jekyll/_cci2/collect-test-data.md [hunk 1]
```diff
@@ -75,102 +75,59 @@ Write the XML files to a subdirectory if you have a custom test step that produc
...
-            find . -type f -regex ".*/build/test-results/.*xml" -exec cp {} ~/test-results/junit/ \;
-          when: always
-      - store_test_results:
-          path: ~/test-results
-      - store_artifacts:
-          path: ~/test-results/junit
-```
+For more details on `--runInBand`, refer to the [Jest CLI](https://facebook.github.io/jest/docs/en/cli.html#runinband) documentation. For more information on these issues, see [Issue 1524](https://github.com/facebook/jest/issues/1524#issuecomment-262366820) and [Issue 5239](https://github.com/facebook/jest/issues/5239#issuecomment-355867359) of the official Jest repository.
 
 #### <a name="mochajs"></a>Mocha for Node.js
```
Comment (devinmarieb) [inline jekyll/_cci2/collect-test-data.md:132]: This section is now blank? There doesn't appear to be any example of Mocha for Node.js

File: jekyll/_cci2/collect-test-data.md [hunk 2]
```diff
@@ -75,102 +75,59 @@ Write the XML files to a subdirectory if you have a custom test step that produc
...
 {: #lessa-namemochajsgreaterlessagreatermocha-for-nodejs }
```
Comment (devinmarieb) [inline jekyll/_cci2/collect-test-data.md:133]: What is this? This seems to be a larger issue on this page where links are getting wrapped in some kind of external link div, when there is nothing clickable, rather it's just a copy for the URL. But this URL is strange to me.

**Review by devinmarieb — CHANGES_REQUESTED** [review]

File: jekyll/_cci2/collect-test-data.md (hunk 1 above)
Comment (devinmarieb) [reply jekyll/_cci2/collect-test-data.md:132]: I'm not quite sure what's happening here. Your preview link shows content for this section, but the Github preview is blank except for the header of Mocha for Node.js

Comment (circlecai) [reply jekyll/_cci2/collect-test-data.md:132]: What is the Github preview? I see content both on my local dev and viewing the file on github, but you probably mean something else.

Comment (devinmarieb) [reply jekyll/_cci2/collect-test-data.md:132]: @circlecai Sorry, this is all good. We identified some errors on this page yesterday (that have nothing to do with your PR) and those errors caused some confusion on my end. We've sorted out how to fix it though. Thanks! I will merge this now.

**Review by devinmarieb — APPROVED** [review]

**Pull request merged**